lib/manageiq/api/open_api/generator.rb

```diff
@@ -32,26 +56,521 @@ def api_version
+                "tags" => ["Authentication"]
+              },
+              "delete" => {
+                "summary" => "Logout",
+                "description" => "Logout current user and invalidate session",
+                "responses" => {
+                  "204" => {"description" => "Logout successful"}
+                },
+                "tags" => ["Authentication"]
+              }
+            }
+          }
+        end
+
+        # Helper methods for checking collection/resource capabilities
+        def supports_collection_action?(collection, action)
+          collection.respond_to?(:collection_actions) && 
+          collection.collection_actions&.key?(action)
+        end
 
+        def supports_resource_action?(collection, action)
+          collection.respond_to?(:resource_actions) && 
+          collection.resource_actions&.key?(action)
+        end
+
+        def get_resource_actions(collection)
+          if collection.respond_to?(:resource_actions) && collection.resource_actions
+            collection.resource_actions.keys
+          else
+            []
+          end
+        end
+
+        # Build schema for create operations (excludes read-only fields)
+        def build_create_schema(model_class, schema_name)
+          {
+            "type" => "object",
+            "properties" => build_writable_properties(model_class),
+            "additionalProperties" => false
+          }
+        end
+
+        # Build schema for update operations (all fields optional, excludes read-only)
+        def build_update_schema(model_class, schema_name)
+          properties = build_writable_properties(model_class)
+          # Make all properties optional for PATCH
+          properties.each { |_, prop| prop.delete("required") if prop.is_a?(Hash) }
+          
+          {
+            "type" => "object",
+            "properties" => properties,
+            "additionalProperties" => false
+          }
+        end
+
+        # Get writable properties (exclude read-only fields like id, timestamps)
+        def build_writable_properties(model_class)
+          read_only_fields = %w[id created_at updated_at created_on updated_on]
+          
+          model_class.columns_hash.each_with_object({}) do |(key, value), properties|
+            next if read_only_fields.include?(key)
+            next if key.ends_with?('_id') && key != model_class.primary_key
+            
+            properties[key] = build_schema_properties_value(model_class, key, value)
+          end
+        end
+
+        # Build comprehensive common parameters
+        def build_common_parameters
+          {
+            "id" => {
+              "name" => "id",
+              "in" => "path",
+              "description" => "ID of the resource",
+              "required" => true,
+              "schema" => {"$ref" => "##{SCHEMAS_PATH}/ID"},
+              "example" => "123"
+            },
+            "subcoll_id" => {
+              "name" => "subcoll_id",
```

Comment:
I think this should be spelled out here.
```suggestion
            "subcollection_id" => {
              "name" => "subcollection_id",
```